src/quacc/atoms/skzcam.py

```diff
@@ -1,73 +1,166 @@
 from __future__ import annotations
 
 from pathlib import Path
-from typing import Dict, Union
+from typing import TYPE_CHECKING
 
 import numpy as np
-from ase import Atoms, io
+from ase.atoms import Atoms
+from ase.io import write
 from ase.units import Bohr
 
+if TYPE_CHECKING:
+    from numpy.typing import NDArray
 
-def convert_pun_to_Atoms(
-    pun_filename: Union[str, Path], atom_oxi_states: Dict
+
+def create_skzcam_clusters(
+    pun_file: str | Path,
+    center_position: NDArray,
+    atom_oxi_states: dict[str, float],
+    shell_max: int = 10,
+    shell_width: float = 0.005,
+    bond_dist: float = 2.5,
+    write_clusters: bool = False,
+    write_clusters_path: str | Path = ".",
+    write_include_ecp: bool = False,
+) -> list[list[int]]:
+    """
+    Returns a list of list containing the indices of the atoms (in embedded_cluster) which form the quantum clusters in the SKZCAM protocol. The number of clusters created is controlled by the rdf_max parameter.
+
+    Parameters
```

Comment:
TODO: A bunch of the docstring parameters here are not quite right.

Reply:
I have now updated the docstrings to reflect the actual list of parameters of the function.